FirebaseTestProject/Views/PostRow.swift

```diff
@@ -8,25 +8,92 @@
-             Text(post.text)
-                 .font(.body)
-                 .padding(.bottom, 8)
-             Text(post.author)
-                 .padding(.bottom, 8)
-         }
-   }
+    var body: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            HStack {
+                Text(post.title)
+                    .font(.largeTitle)
+                    .padding(.leading)
+                Spacer()
+                let userid = UserDefaults.standard.value(forKey: "userid") != nil ? UUID(uuidString: UserDefaults.standard.value(forKey: "userid") as! String) : UUID(uuidString: "00854E9E-8468-421D-8AA2-605D8E6C61D9")
+                if userid == post.authorid {
+                    Button(action: {
+                        showAlert = true
+                    }, label: {
+                        Label("", systemImage: "trash")
+                    })
+                    .buttonStyle(PlainButtonStyle())
+                }
+                Button(action: {
+                    Task {
+                        post.isFavorite = !post.isFavorite
+                        post.isFavorite ?
+                            try await PostService.favorite(post) : try await PostService.delete(favorite: post)
+                        isFavoriteAction(post)
+                    }
+                }, label: {
+                    Label("", systemImage: post.isFavorite ? "heart.fill" : "heart")
+                })
+                .padding()
+                .buttonStyle(PlainButtonStyle())
+            }
+            .alert(isPresented: $showAlert) {
+                Alert(title: Text("Do you want to delete this post?"),
+                      primaryButton: .cancel(),
+                      secondaryButton: .destructive(Text("Delete")) {
+                    deletePostAction(post)
+                })
+            }
+            HStack {
+                Text(post.author)
+                    .padding()
+                    .foregroundColor(Color(uiColor: UIColor.systemBlue))
+                    .onTapGesture {
+                        isPresenting = true
+                    }.sheet(isPresented: $isPresenting, onDismiss: {
+                        isPresenting = false
+                    }) {
+                        PostsList(viewStyle: .singleAuthor(post.author))
+                    }
+                Spacer()
+                Text(DateFormatter.postFormat(date: post.timestamp))
+            }
+            .padding([.leading, .trailing], 20)
+            Divider()
+            Text(post.text)
+                .font(.body)
+                .padding([.bottom, .top])
+                .padding(.leading, 30)
+                // Accomplishes Auto-Height for Multi-Text
+                // It expands the view to a fixed size Vertically, but maintains the stature of
+                // the horizontal component
+                .fixedSize(horizontal: false, vertical: true)
+        }
+        .background(Color.blue.opacity(0.5))
```

Comment:
I would remove the blue background for now. From an accessibility standpoint, blue buttons on a blue background might lead to insufficient contrast between UI elements.

<img src="https://user-images.githubusercontent.com/34844819/130521444-d09b49ea-cd3f-4305-9610-dffd8520c3f3.png" width="300px" />